src/utilcode/winwrappers.cpp

```diff
@@ -0,0 +1,996 @@
+                    Newpath.GetUnicode(),
+                    lpProgressRoutine,
+                    lpData,
+                    pbCancel,
+                    dwCopyFlags
+                    );
+            
+            lastError = GetLastError();
+        }
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+}
+
+HANDLE
+FindFirstFileExWrapper(
+        _In_ LPCWSTR lpFileName,
+        _In_ FINDEX_INFO_LEVELS fInfoLevelId,
+        _Out_writes_bytes_(sizeof(WIN32_FIND_DATAW)) LPVOID lpFindFileData,
+        _In_ FINDEX_SEARCH_OPS fSearchOp,
+        _Reserved_ LPVOID lpSearchFilter,
+        _In_ DWORD dwAdditionalFlags
+        )
+{
+    HRESULT hr = S_OK;
+    HANDLE ret = INVALID_HANDLE_VALUE;
+    DWORD lastError;
+    EX_TRY
+    {
+        Win32StackSString path(Win32StackSString::Literal, lpFileName);
+
+        if (SUCCEEDED(LongFile::NormalizePath(path)))
+        {
+            ret = FindFirstFileExW(
+                    path.GetUnicode(),
+                    fInfoLevelId,
+                    lpFindFileData,
+                    fSearchOp,
+                    lpSearchFilter,
+                    dwAdditionalFlags
+                    );
+
+            lastError = GetLastError();
+        }
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == INVALID_HANDLE_VALUE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+}
+#endif //!FEATURE_PAL
+
+#ifndef FEATURE_PAL
+SString LongFile::ExtendedPrefix(SString::Literal, W("\\\\?\\"));
+SString LongFile::DevicePathPrefix(SString::Literal, W("\\\\.\\"));
+SString LongFile::UNCExtendedPathPrefix(SString::Literal, W("\\\\?\\UNC\\"));
+SString LongFile::UNCPathPrefix(SString::Literal, W("\\\\"));
+
+const WCHAR LongFile::VolumeSeparatorChar = W(':');
+const WCHAR LongFile::DirectorySeparatorChar = W('\\');
+const WCHAR LongFile::AltDirectorySeparatorChar = W('/');
+
+BOOL LongFile::IsExtended(SString & path)
+{
+    return path.BeginsWith(ExtendedPrefix);
+}
+
+BOOL LongFile::IsUNCExtended(SString & path)
+{
+
+    return path.BeginsWith(UNCExtendedPathPrefix);
+}
+
+BOOL LongFile::IsDirectorySeparator(WCHAR c)
+{
+    return c == DirectorySeparatorChar || c == AltDirectorySeparatorChar;
+}
+
+BOOL LongFile::IsRelative(SString & path)
+{
+    if (path.GetCount() < 2) 
+    { 
+        return TRUE;  // It isn't fixed, it must be relative.  There is no way to specify a fixed path with one character (or less).
+    }
+
+    if (IsDirectorySeparator(path[0]))
+    {
+        return !IsDirectorySeparator(path[1]); // There is no valid way to specify a relative path with two initial slashes
+    }
+
+    return !((path.GetCount() >= 3)           //The only way to specify a fixed path that doesn't begin with two slashes is the drive, colon, slash format- i.e. C:\
```

Comment:
What if the path is something like "c:\foo\bar\..\zoo\file.cpp"? This is relative path as well and this function is not handling such cases.